File: app/src/main/java/com/platdmit/simplecloudmanager/app/adapters/DomainRecordListAdapter.java
package com.platdmit.simplecloudmanager.app.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import com.platdmit.simplecloudmanager.R;
import com.platdmit.simplecloudmanager.domain.models.DomainRecord;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class DomainRecordListAdapter extends RecyclerView.Adapter<DomainRecordListAdapter.DomainRecordListHolder> {

    private List<DomainRecord> elementList;

    @NonNull
    @Override
    public DomainRecordListAdapter.DomainRecordListHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        int layoutType = R.layout.fragment_drecord_list_item;
        return new DomainRecordListHolder(layoutInflater, parent, layoutType);
    }

    @Override
    public void onBindViewHolder(@NonNull DomainRecordListHolder holder, int position) {
        holder.bindData(elementList.get(position));
    }

    @Override
    public int getItemCount() {
        return elementList.size();
    }

    public void setContentData(List<DomainRecord> elements){
        this.elementList = elements;
    }

    class DomainRecordListHolder extends RecyclerView.ViewHolder{

        private TextView mHost = itemView.findViewById(R.id.record_host);
        private TextView mType = itemView.findViewById(R.id.record_type);
        private TextView mTtl = itemView.findViewById(R.id.record_ttl);
        private TextView mPriority = itemView.findViewById(R.id.record_priority);
        private TextView mData = itemView.findViewById(R.id.record_data);

        public DomainRecordListHolder(LayoutInflater inflater, ViewGroup parent, int layoutType) {
            super(inflater.inflate(layoutType, parent, false));
        }

        public void bindData(DomainRecord data){
            mHost.setText(data.getName());
            mType.setText(data.getType());
            mData.setText(data.getData());
            mPriority.setText(data.getPriority());
            mTtl.setText(data.getTtl());
        }
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/screens/domain/DomainRecordListAdapter.kt
package com.platdmit.simplecloudmanager.screens.domain

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.platdmit.domain.models.DomainRecord
import com.platdmit.simplecloudmanager.databinding.FragmentDrecordListItemBinding
import com.platdmit.simplecloudmanager.screens.domain.DomainRecordListAdapter.DomainRecordListHolder

class DomainRecordListAdapter : RecyclerView.Adapter<DomainRecordListHolder>() {
    private val elementList: MutableList<DomainRecord> = mutableListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DomainRecordListHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val viewBinding = FragmentDrecordListItemBinding.inflate(layoutInflater, parent, false)
        return DomainRecordListHolder(viewBinding)
    }

    override fun onBindViewHolder(holder: DomainRecordListHolder, position: Int) {
        holder.bindData(elementList[position])
    }

    override fun getItemCount(): Int {
        return elementList.size
    }

    fun setContentData(elements: List<DomainRecord>) {
        elementList.clear()
        elementList.addAll(elements)
        notifyDataSetChanged()
    }

    inner class DomainRecordListHolder(
            private val viewBinding: FragmentDrecordListItemBinding
    ) : RecyclerView.ViewHolder(viewBinding.root) {
        fun bindData(data: DomainRecord) {
            viewBinding.run {
                recordHost.text = data.name
                recordType.text = data.type
                recordData.text = data.data
                recordPriority.text = data.priority
                recordData.text = data.ttl
            }
        }
    }
}